### PR TITLE
Fix error that signal handling triggers Thread Sanitizer warning

### PR DESCRIPTION
**Pull request by YuzhouGuo**
Associated issue
- https://github.com/cyfitech/cris/issues/921

According to [the glog reference](https://rpg.ifi.uzh.ch/docs/glog.html)

> Raw Logging
> **The header file <glog/raw_logging.h> can be used for thread-safe logging**, which does not allocate any memory or acquire any locks. Therefore, the macros defined in this header file can be used by low-level memory allocation and synchronization code. Please check src/glog/raw_logging.h.in for detail.

### CODE REVIEW

File: src/signal/cr_signal.cc [hunk 1]
```diff
@@ -9,6 +9,8 @@
 #pragma GCC diagnostic ignored "-Wshadow"
 #endif
 
+#include <glog/raw_logging.h>
```
[the remark this answers is not in the document]
Comment (YuzhouGuo) [reply src/signal/cr_signal.cc:12]: tidy自动做的排序结果就是这样

File: src/signal/cr_signal.cc [hunk 2]
```diff
@@ -9,8 +9,12 @@
...
+
 #include <boost/stacktrace.hpp>
 
+#include <unistd.h>
```
Comment (chenhao94) [inline src/signal/cr_signal.cc:16]: 这个我们在哪用到的？

**Review by chenhao94 — APPROVED** [review]

File: src/signal/cr_signal.cc [hunk 3]
```diff
@@ -52,20 +55,27 @@ static void InvokeDefaultSignalHandler(int signal_number) {
...
-               << siginfo->si_addr << ") ****";
-
-    LOG(ERROR) << "PID " << getpid() << ", TID " << std::this_thread::get_id() << ", from PID " << siginfo->si_pid;
+    RAW_LOG(ERROR, "******************************************************");
+
+    RAW_LOG(
+        ERROR,
+        "**** Signal %d (%s) received (@%p) ****",
+        signal_number,
+        strsignal(signal_number),
```
[the remark this answers is not in the document]
Comment (chenhao94) [reply src/signal/cr_signal.cc]: @zptan `sigdescr_np` 是posix的吗？我不确定非linux系统下有没有这个东西。只用数值我觉得也行。

@YuzhouGuo 如果你要用strsignal记得检查返回值

https://linux.die.net/man/3/strsignal
> The strsignal() function returns the appropriate description string, or an unknown signal message if the signal number is invalid. On some systems (but not on Linux), a NULL pointer may be returned instead for an invalid signal number.

[the remark this answers is not in the document]
Comment (chenhao94) [reply src/signal/cr_signal.cc]: 其他的posix或者没有新glibc的时候是怎么解决这个问题的？

https://pubs.opengroup.org/onlinepubs/9699919799/functions/strsignal.html
> The strsignal() function need not be thread-safe.

说明他们没觉得thread safety是个问题？

File: src/signal/cr_signal.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (chenhao94) [reply src/signal/cr_signal.cc:12]: 我们对不同第三方库给了不同的顺序，基本上越底层越排后面
https://github.com/cyfitech/cris-core/blob/d7506534c0d013c189b496e362fc4e9c04c54b90/.clang-format#L39-L78

File: src/signal/cr_signal.cc (hunk 2 above)
Comment (xkszltl) [reply src/signal/cr_signal.cc:16]: Avoid POSIX-only header if possible for best portability.

Comment (YuzhouGuo) [reply src/signal/cr_signal.cc:16]: `getpid()`
According to [the reference](https://man7.org/linux/man-pages/man2/getpid.2.html)

Comment (chenhao94) [reply src/signal/cr_signal.cc:16]: pid还是需要的

Comment (xkszltl) [reply src/signal/cr_signal.cc:16]: 我记得是不是thread本身有native handler之类的就是pid？

**Pull request merged**